test/fixtures/suit-nesting.css

```diff
@@ -0,0 +1,33 @@
+/** @define Component **/
+
+.Component {
+  color: green;
+
+  .Component-descendant {}
+  &.is-stateName {}
+  &-descendant {}
+  &-descendant--modifier {}
+  &--modifier {}
+  &:pseudo-class {}
+  &:pseudo-element {}
+  &::pseudo-element {}
+  &[attr] {}
+}
+
+.Component-standAloneDescendant {}
+
+.Component {
```

Comment:
Ah ok, fine with me with fixing the SUIT patterns only.
The following should be invalid in SUIT:

```css
.Component {
  .Component-descendant {}
}
```